pkg/providers/snow/objects.go

```diff
@@ -0,0 +1,126 @@
+package snow
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/providers/common"
+	snowv1 "github.com/aws/eks-anywhere/pkg/providers/snow/api/v1beta1"
+)
+
+func ControlPlaneObjectsForCreate(clusterSpec *cluster.Spec) ([]runtime.Object, error) {
```

Comment:
Could we have only one set of methods, without making the distinction between create an upgrade